dashboard/app/helpers/aitutor_system_prompt_helper.rb

```diff
@@ -0,0 +1,81 @@
+      Overall Objective: This comprehensive framework aims to cultivate a focused, respectful, and enriching educational dialogue within a
+      computer science classroom, prioritizing student learning, safety, and inclusivity.
+      If you determine the question is relevant to computer science, remember:
+      1. The student prefers being guided to discover the solution themselves over being directly provided with answers.
+      2. The student prefers to receive guided questions that prompt them to reevaluate assumptions or scrutinize overlooked details in their code.
+      3. The student is interested in being reminded of important principles and concepts in programming, which can often be the root cause of issues in their code.
+      Direct Support: Engage exclusively in discussions that directly support the Code.org computer science curriculum,
+      focusing solely on the context of a specific level or lesson. Your job is to provide the student with one next step to take, in priority order:
+      If their code does not compile, give them a hint about why
+      If their tests are failing, give them a hint about why
+      If their solution is missing something, give them a hint about what it's missing.
+      If there is a bug in their code, tell them where the issue is."
+    base_system_prompt
+  end
+
+  def self.get_language_specific_system_prompt(unit)
+    language = unit.csa? ? 'Java' : 'Python'
+    "Specific Exclusions: Refrain from discussing topics not explicitly related to computer
+    science or #{language} programming."
+  end
+
+  def self.get_level_instructions(level)
+    level_instructions = level.properties["long_instructions"]
+    "\n Here are the student instructions for this level: #{level_instructions}"
+  end
+
+  def self.get_validated_level_test_file_contents(level)
+    test_file_contents = ""
+    if level.validation
+      if level.validation.values.empty?
+        return render(status: :bad_request, json: {message: "There are no test files associated with level id=#{level_id}."})
+      else
+        level.validation.each_value do |validation|
+          test_file_contents += validation["text"]
+        end
+      end
+    end
+    test_file_contents.empty? ?
+      "\n There are no tests for this level." :
+      "\n The contents of the test file are: #{test_file_contents}"
+  end
+
+  def self.get_system_prompt(level_id, script_id)
+    level = Level.find(level_id)
+    unless level
+      return render(status: :bad_request, json: {message: "Couldn't find level with id=#{level_id}."})
+    end
+
+    unit = Unit.find(script_id)
+    unless unit
+      return render(status: :bad_request, json: {message: "Couldn't find unit with id=#{script_id}."})
+    end
+
+    system_prompt =
+      get_base_system_prompt +
+      get_language_specific_system_prompt(unit) +
+      get_level_instructions(level) +
+      get_validated_level_test_file_contents(level)
```

Comment:
I like this set up because it makes it super easy to also add in level-specific customizations if we store them as properties on a level in the future.